modules/core/src/convert.cpp

```diff
@@ -5371,6 +5371,52 @@ static bool ocl_convertScaleAbs( InputArray _src, OutputArray _dst, double alpha
     return k.run(2, globalsize, NULL, false);
 }
 
+static bool ocl_convertFp16( InputArray _src, OutputArray _dst )
+{
+    int type = _src.type(), depth = CV_MAT_DEPTH(type), cn = CV_MAT_CN(type);
+
+    int nSChannels = _src.getMat().channels();
+    int nSDims = _src.getMat().dims;
+
+    int ddepth = 0;
+
+    switch( depth )
+    {
+    case CV_32F:
+        ddepth = CV_16S;
+        break;
+    case CV_16S:
+        ddepth = CV_32F;
+        break;
+    default:
+        CV_Error(Error::StsUnsupportedFormat, "Unsupported input depth");
```

Comment:
Probably this check should be before `CV_OCL_RUN` call.
Arguments check code is common in general, so it should not be duplicated between "implementations".